### DISCUSSION TITLE
power cycle/reset compute board + watchdog

### DISCUSSION THREAD

**Discussion by db-voxel** [iRobotEducation/create3_docs#331] [Ideas]
It would be very helpful to have an option for an autonomous power cycle or reset of the compute board.
1. A ros service for immediate reset/power cycle.
2. A ros watchdog service for timed out reset unless the compute board strobes the timer.
3. A button in the web interface to reset.power cycle the compute board.

**Comment by shamlian** [comment]
I'm sorry, AFAIK it's not possible to power-cycle the USB-C port through software. I will add the request to our backlog for further investigation.

**Reply by db-voxel** [reply]
Hi @shamlian 
I believe the power cycle can be done with the port's power removal/re-application.
If USB/PD is implemented on the adapter board it is possible via a TCPC. Otherwise, the usb power can be controlled e.g. via gating power to USB phy/host controller.

**Reply by shamlian** [reply]
The Create 3 robot adapter board implements passive downstream-facing Rp pull-ups to advertise a 3.0 A, 5 V capability to devices. The power is switched on when Rd is detected from a downstream device via a comparator. The power is switched off in the event of an overcurrent (or disconnect). There is no PD chip in the loop. To my knowledge, the controller does not have the ability to remove power from the port. This is something we could investigate in a future hardware iteration.

**Reply by db-voxel** [reply]
Thanks for the explanation.
Is it then possible to implement the whole system power cycle, i.e. the adapter board along with the compute board?
